src/software/sensor_fusion/refbox_data.h

```diff
@@ -0,0 +1,259 @@
+        : packet_timestamp(packet_timestamp),
+          game_state_timestamp(game_state_timestamp),
+          game_state_counter(game_state_counter),
+          designated_position(designated_position),
+          blue_team_on_positive_half(blue_team_on_positive_half),
+          current_game_state_time_remaining(current_game_state_time_remaining),
+          us_team_info(us_team_info),
+          them_team_info(them_team_info),
+          game_state(game_state),
+          next_game_state(next_game_state),
+          stage(stage),
+          game_events(game_events),
+          proposed_game_events(proposed_game_events)
+    {
+    }
+
+    Timestamp getPacketTimestamp(void)
+    {
+        return packet_timestamp;
+    }
+
+    Timestamp getGameStateTimestamp(void)
+    {
+        return game_state_timestamp;
+    }
+
+    int getGameStateCounter(void)
+    {
+        return game_state_counter;
+    }
+
+    Point getDesignatedPosition(void)
+    {
+        return designated_position;
+    }
+
+    bool getBlueTeamOnPositiveHalf(void)
+    {
+        return blue_team_on_positive_half;
+    }
+
+    Duration getCurrentGameStateTimeRemaining(void)
+    {
+        return current_game_state_time_remaining;
+    }
+
+    TeamInfo getUsTeamInfo(void)
+    {
+        return us_team_info;
+    }
+
+    TeamInfo getThemTeamInfo(void)
+    {
+        return them_team_info;
+    }
+
+    RefboxGameState getGameState(void)
+    {
+        return game_state;
+    }
+
+    RefboxGameState getNextGameState(void)
+    {
+        return next_game_state;
+    }
+
+    RefboxStage getStage(void)
+    {
+        return stage;
+    }
+
+    std::vector<GameEvent> getGameEvents(void)
+    {
+        return game_events;
+    }
+
+    std::vector<ProposedGameEvent> getProposedGameEvents(void)
+    {
+        return proposed_game_events;
+    }
+
+
+   private:
+    // packet timestamp
+    Timestamp packet_timestamp;
+    // time that current game state was initiated
+    Timestamp game_state_timestamp;
+    // number of game states issued since startup (mod 2^32)
+    int game_state_counter;
+
+    // ball placement position
+    Point designated_position;
+    // whether or not blue team is on positive half
+    bool blue_team_on_positive_half;
+    // time remaining for current game_state
+    Duration current_game_state_time_remaining;
+
+    // Referee's info on our team
+    TeamInfo us_team_info;
+    // Referee's info on their team
+    TeamInfo them_team_info;
+    // Current game state
+    RefboxGameState game_state;
+    // The game state that will be issued after the current stoppage and ball placement to
+    // continue the game.
+    RefboxGameState next_game_state;
+    // Current stage of play
+    RefboxStage stage;
+
+    // Note: we're using (Proposed)GameEvent proto
+    // directly instead of wrapper because YAGNI
+    // We don't see a use for these events currently
+    // but they must be included for completeness
```

Comment:
Our internal classes should not expose interfaces returning data types external to our system unless absolutely needed. Could we remove the getter for this?

Reply:
fixed